updatecli/scripts/check-jdk.sh

```diff
@@ -30,10 +30,10 @@ function get_jdk_download_url() {
       echo "https://github.com/adoptium/temurin19-binaries/releases/download/jdk-${jdk_version}/OpenJDK19U-jdk_${platform}_hotspot_${jdk_version//+/_}";
       return 0;;
     21*)
-      # JDK version (21+35-ea-beta)
-      ##    https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35-ea-beta/OpenJDK21U-jdk_aarch64_linux_hotspot_ea_21-0-35.tar.gz
+      # JDK version (21+35)
+      ##    https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35/OpenJDK21U-jdk_aarch64_linux_hotspot_21_35.tar.gz
       urlEncodedJDKVersion="${jdk_version//+/%2B}"
```

Comment:
```suggestion
      underscoredJDKVersion="${jdk_version//+/_}"
```

Reply:
I remember weird issues when trying to download from the URL with the `_`. Let's keep the encoded version (unless you are able to test your suggestion and prove that it works)